dataframely/schema.py

```diff
@@ -628,6 +631,191 @@ def serialize(cls) -> str:
         }
         return json.dumps(result, cls=SchemaJSONEncoder)
 
+    # ------------------------------------ PARQUET ----------------------------------- #
+
+    @classmethod
+    def write_parquet(
+        cls, df: DataFrame[Self], /, file: str | Path | IO[bytes], **kwargs: Any
+    ) -> None:
+        """Write a typed data frame with this schema to a parquet file.
+
+        This method automatically adds a serialization of this schema to the parquet
+        file as metadata. This metadata can be leveraged by :meth:`read_parquet` and
+        :meth:`scan_parquet` for more efficient reading, or by external tools.
+
+        Args:
+            df: The data frame to write to the parquet file.
+            file: The file path or writable file-like object to which to write the
+                parquet file. This should be a path to a directory if writing a
+                partitioned dataset.
+            kwargs: Additional keyword arguments passed directly to
+                :meth:`polars.write_parquet`. ``metadata`` may only be provided if it
+                is a dictionary.
+
+        Attention:
+            Be aware that this method suffers from the same limitations as
+            :meth:`serialize`.
+        """
+        metadata = kwargs.pop("metadata", {})
+        df.write_parquet(
+            file, metadata={**metadata, _METADATA_KEY: cls.serialize()}, **kwargs
+        )
+
+    @classmethod
+    def sink_parquet(
+        cls, lf: LazyFrame[Self], /, file: str | Path | IO[bytes], **kwargs: Any
```

Comment:
Polars supports passing a `PartitioningScheme` object here, e.g., [`PartitionParted`](https://docs.pola.rs/api/python/dev/reference/api/polars.PartitionParted.html) or [`PartitionByKey`](https://docs.pola.rs/api/python/dev/reference/api/polars.PartitionByKey.html#polars.PartitionByKey). Should we also support this? I think it's quite useful to write hive-partitioned parquets.

Reply:
Agreed; it's a little annoying because `PartitioningScheme` is only defined in `polars._typing` (private module), but let's go for it nonetheless.